cosmic/src/hrdiag.f

```diff
@@ -659,18 +660,26 @@ SUBROUTINE hrdiag(mass,aj,mt,tm,tn,tscls,lums,GB,zpars,
                      mc = mt
                   endif
 
-                  if(mt.le.mxns)then
-*
-* Zero-age Neutron star
-*
+* Specify the baryonic to gravitational remnant mass prescription
+* MJZ 04/2020
+
+* Determine gravitational mass using Lattimer & Yahil 1989 for nsflag>1
+                  if(nsflag.le.1)then
+                     mrem = mt
+                  else
+                     mrem = 6.6666667d0*(SQRT(1.d0+0.3d0*mt)-1.d0)
+* If rembar_massloss >= 0, limit the massloss by rembar_massloss
+                     if(rembar_massloss.ge.0d0)then
+                        if((mt-mrem).ge.rembar_massloss) 
+     &                               mrem = mt-rembar_massloss
+                     endif
+                  endif
+
+* Determine whether a zero-age NS or BH is formed
+                  if(mrem.le.mxns)then
+                     mt = mrem
                      kw = 13
-* Convert baryonic mass to gravitational mass (Lattimer & Yahil 1989)
-                     if(nsflag.ge.2) mt = 2.108167d0*SQRT(10.d0+3.d0*mt)
```

Comment:
The deleted code for mt doesn't seem to match mrem on line 670. Specifically, the SQRT(10d.0 + 3.d0*mt) in 668 and the  SQRT(1.d0 + 0.3*mt) 

I'm probably missing something but this is probably worth a quick check.

Reply:
The full expression in the old version is:  
668                      if(nsflag.ge.2) mt = 2.108167d0*SQRT(10.d0+3.d0*mt)
669      &                                    - 6.6666667d0

which is equivalent to (but more complicated then): 
6.6666667d0*(SQRT(1.d0+0.3d0*mt)-1.d0)

Reply:
oh youre totally right; I was dividing instead of multiplying. sorry!